### DISCUSSION TITLE
Send value of counter1 to mqtt upon change

### DISCUSSION THREAD

**Discussion by jeroenst** [arendst/Tasmota#19550] [Q&A] [answered]
I want the value of counter1 to be send to mqtt upon change.

I tried to use this rule without succes:

Rule1 ON %mem1%!=counter#c1 DO Backlog mem1 %value%; publish home/TASMOTA-WATERMETER/counter1 %mem1% ENDON

What am I doing wrong?

**Comment by jeroenst** [comment, marked as answer]
Rule1 ON counter#c1!=%var1% DO Backlog var1 %value%; publish2 home/TASMOTA-WATERMETER/counter1 %value% ENDON 

works

**Reply by sfromis** [reply]
Triggers are **always** with the dependent condition first, and then variables are supported to compare to. Beware that as a starting point, your usage of `mem1` causes flash write every time, potentially a lot of flash wear. And if you want to publish the current value, use `%value%` in the publish, otherwise you'll get the previous value. You cannot set and then use a variable with `%` in the same rule.

**Reply by jeroenst** [reply]
Thank you very much!
